include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -261,26 +276,39 @@ class is_operational_impl
                 if (check_existence_of_kinks_in_input_wires(given_cds, input_pattern) ||
                     check_existence_of_kinks_in_output_wires(given_cds, input_pattern))
                 {
-                    return operational_status::NON_OPERATIONAL;
+                    non_operational_due_to_kinks = true;
                 }
             }
         }
 
+        if (non_operational_due_to_kinks)
+        {
+            return operational_status::NON_OPERATIONAL;
+        }
+
         // if we made it here, the layout is operational
         return operational_status::OPERATIONAL;
     }
     /**
-     * Determines the input combinations yielding the correct output.
+     * Determines the input combinations yielding the `non-operational` status due
+     * to kinks. `True` indicates that the layout is `non-operational` due to kinks, `false` otherwise.
      *
-     * @return All inputs (e.g. 2-input Boolean function: 00 ^= 0; 10 ^= 2) for which the correct output is computed.
+     * @return Set of pairs where the first element is the input pattern (e.g. 2-input Boolean function: 00 ^= 0;
+     * 10 ^= 2) for which the wrong output is computed. The second entry indicates the information if the
+     * `non-operational` status is due to kinks. `True` indicates that the layout is `non-operational` due to kinks,
+     * `false` otherwise.
+     *
+     * @note "Non-operational due to kinks" refers to the operational status being exclusively caused to be
+     * `NON_OPERATIONAL` due to kinks.
      */
-    [[nodiscard]] std::set<uint64_t> determine_operational_input_patterns() noexcept
+    [[nodiscard]] std::set<std::pair<uint64_t, bool>>
+    determine_non_operational_input_patterns_and_op_status_due_kinks() noexcept
```

Comment:
An operational status should not be a `bool` but an `operational_status`

Reply:
I know that is not great and confusing.
The reason is that ``operational_status`` is not great either. I think I need to get rid of ``_op_status_due_`` in the description.

Reply:
yeah, it is wrong to call it ``op_status_...``. It is something else. I agree, we should represent this by a new sata type. Thanks!